application/frontend/controllers/AuthController.php

```diff
@@ -128,6 +164,65 @@ public function getAfterLoginUrl($returnTo)
         } else {
             $path = '';
         }
-        return \Yii::$app->params['ui_url'] . $path;
+        return \Yii::$app->params['uiUrl'] . $path;
+    }
+
+    /**
+     * Get client_id from request or session and then store in session
+     * @return string
+     * @throws BadRequestHttpException
+     */
+    public function getClientIdOrFail()
+    {
+        $clientId = \Yii::$app->request->get('client_id');
+        if ($clientId === null) {
+            $clientId = \Yii::$app->session->get('clientId');
+            if ($clientId === null) {
+                throw new BadRequestHttpException('Missing client_id');
```

Comment:
Does the `'Missing client_id'` text get returned to the requesting browser? If so, do we want to tell users making invalid requests why their request was invalid? It seems like invalid requests will (almost always) be coming from users trying to hack the system, rather than users using it the normal way, in which case it seems better to not give them any helpful clues.